docs/azure/role-assignments.md

```diff
@@ -0,0 +1,156 @@
+---
+title: Add Azure role assignments
+description: Learn how to add Azure role assignments to .NET Aspire resources.
+ms.date: 03/24/2025
+---
+
+# Add Azure role assignments
+
+All .NET Aspire Azure hosting integrations define Azure resources. [These resources](integrations-overview.md#add-azure-resources) come with default role assignments. You can replace these default role assignments with built-in role or custom role assignments. In this article, you learn how to add Azure role assignments to .NET Aspire resources.
+
+## Add built-in role assignments
+
+For every Azure resource you add to the [app model](xref:dotnet/aspire/app-host#terminology), you can add built-in role assignments. When a resource is referenced by another resource, a dependency is created. The dependent resource has default role access that's defined by each Azure resource When an API project resource references an Azure resource, default role assignments are created. You can override these defaults and assign built-in roles that corresponding to the resource type. Consider the following example:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var search = builder.AddAzureSearch("search");
+
+var api = builder.AddProject<Projects.Api>("api")
+                 .WithReference(search);
+```
+
+In the preceding code, the `search` resource is referenced by the `api` resource. The `search` resource has default role assignments:
+
+- <xref:Azure.Provisioning.Search.SearchBuiltInRole.SearchIndexDataContributor?displayProperty=nameWithType>
+- <xref:Azure.Provisioning.Search.SearchBuiltInRole.SearchServiceContributor?displayProperty=nameWithType>
+
+To override these defaults and assign built-in roles that correspond to the resource type. For example:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var search = builder.AddAzureSearch("search");
+
+var api = builder.AddProject<Projects.Api>("api")
+                 .WithRoleAssignments(search, SearchBuiltInRole.SearchIndexDataReader)
+                 .WithReference(search);
+```
+
+The preceding code replaces the default role assignments with the following:
+
+- <xref:Azure.Provisioning.Search.SearchBuiltInRole.SearchIndexDataReader?displayProperty=nameWithType>
+
+This limits the API project to only read data from the Azure Search resource. The API project can no longer write data to the Azure Search resource.
+
+When you chain a call to the `WithRoleAssignments` method, it overrides the default role assignments, replacing them with the given role assignments. The `WithRoleAssignments` method takes the resource to which the role assignment applies and the built-in role to assign. In this case, the `search` resource is assigned the `SearchIndexDataReader` role.
```

Comment:
Do you need a TODO here for API linking?

Also, I feel like this would be better placed at the start of talking about changing the defaults. I don't think you need the last sentence as it really just reiterates what was previously said above:

```markdown
The preceding code replaces the default role assignments with the following:

- <xref:Azure.Provisioning.Search.SearchBuiltInRole.SearchIndexDataReader?displayProperty=nameWithType>
```

Reply:
Done